banyan/resource_service.go

```diff
@@ -1610,39 +1557,7 @@ func resourceServiceRead(ctx context.Context, d *schema.ResourceData, m interfac
 		"include_domains":     service.CreateServiceSpec.Metadata.Tags.IncludeDomains,
 	}
 	d.Set("metadatatags", []interface{}{metadatatags})
-	frontendPort, err := strconv.Atoi(service.CreateServiceSpec.Spec.Attributes.FrontendAddresses[0].Port)
-	if err != nil {
-		diagnostics = diag.FromErr(err)
-		return
-	}
-	backendPort, err := strconv.Atoi(service.CreateServiceSpec.Spec.Attributes.FrontendAddresses[0].Port)
-	if err != nil {
-		diagnostics = diag.FromErr(err)
-		return
-	}
-	spec := map[string]interface{}{
-		"attributes": map[string]interface{}{
-			//todo make this be able to handle n frontend addresses
-			"frontend_address": map[string]interface{}{
-				"cidr": service.CreateServiceSpec.Spec.Attributes.FrontendAddresses[0].CIDR,
-				"port": frontendPort,
-			},
-			//todo make this handle n host tag selectors
-			"host_tag_selector": map[string]interface{}{
-				"site_name": service.CreateServiceSpec.Spec.Attributes.HostTagSelector[0],
-			},
-			"tls_sni": service.CreateServiceSpec.Spec.Attributes.TLSSNI,
-		},
-		"backend": map[string]interface{}{
-			"target": map[string]interface{}{
-				"client_certificate": service.CreateServiceSpec.Spec.Backend.Target.ClientCertificate,
-				"name":               service.CreateServiceSpec.Spec.Backend.Target.Name,
-				"port":               backendPort,
-				"tls":                service.CreateServiceSpec.Spec.Backend.Target.TLS,
-				"tls_insecure":       service.CreateServiceSpec.Spec.Backend.Target.TLSInsecure,
-			},
-		},
-	}
+	spec, diagnostics := flattenServiceSpec(service.CreateServiceSpec.Spec)
```

Comment:
Generally using err is by convention and easy to read instead of diagnostics.

Reply:
Terraform has a diag.Diagnostics type which is recommended by HashiCorp for debugging the terraform provider https://learn.hashicorp.com/tutorials/terraform/provider-debug

Reply:
Should there be a check to see if `diagnostics` contains a fatal error and return immediately?